tests/net/icmpv4/src/main.c

```diff
@@ -441,12 +451,12 @@ static void test_icmpv4_send_echo_req(void)
 	}
 
 	if (net_ipv4_input(pkt)) {
-		net_pkt_unref(pkt);
 		zassert_true(false, "Failed to send");
 	}
+	net_pkt_unref(pkt);
```

Comment:
Why was `net_pkt_unref(pkt);` moved here? If the function returned `NET_OK` here (which is not very clearly verified here) the packet shouldn've been consumed by the stack, unreferencing it here would result in packet being freed twice. Same goes for other places.

Reply:
You are right. The packet should be released by the lower layer. 
At first I modify the code just as what you have said. But I found the test cannot pass under CONFIG_ZTEST_SHUFFLE=y, the packets failed to be allocated. If I enlarge the packet number in prj.conf. Then it can pass with CONFIG_ZTEST_SHUFFLE=y.
I wonder in the lower layer (or the icmp stack) the packet is not unrefed. 
To WA this problem, I modify the code like this. Maybe we should take it as a bug?

Reply:
The behavior doesn't seem right, the ICMPv4 layer should've released the packet if it notified that the packet was consumed. If that's not the case we should treat this as a bug indeed.